src/DeviceInput/deviceInputSystem.ts

```diff
@@ -276,7 +282,10 @@ export class DeviceInputSystem implements IDisposable {
             if (kbKey) {
                 kbKey[evt.keyCode] = 1;
                 if (this.onInputChanged) {
-                    this.onInputChanged(DeviceType.Keyboard, 0, evt.keyCode, 0, kbKey[evt.keyCode]);
+                    const eventData: { [k: string]: any } = {};
+                    eventData.key = evt.key;
+
+                    this.onInputChanged(DeviceType.Keyboard, 0, evt.keyCode, 0, kbKey[evt.keyCode], eventData);

```

Comment:
Couldn't it be interesting to know which event triggered the `onInputChanged` (by passing an additional data to the call that would let know the event was a key down/up/blur/...)? Just thinking out loud here.

Reply:
This sounds like it could be worth exploring.  The only thing that could would give me hesitation, it would be how that functionality would play with Babylon Native.